woocommerce/payment-gateway/payment-tokens/class-sv-wc-payment-gateway-payment-token.php

```diff
@@ -538,6 +538,81 @@ private function read( \WC_Payment_Token $core_token ) {
 	}
 
 
+	/**
+	 * Stores the token data in the database.
+	 *
+	 * Stores the token data as a Woocommerce payment token, in the `wp_woocommerce_payment_tokens` table,
+	 * with meta data in the `wp_woocommerce_payment_tokenmeta` table.
+	 *
+	 * @since 5.6.0-dev.1
+	 */
+	public function save() {
+
+		if ( $this->token instanceof \WC_Payment_Token ) {
+			$token = $this->token;
+		} else {
+
+			// see if there is already a token with this ID saved for this customer and gateway
+			// TODO: use `get_user_id` and `get_gateway_id`, once merged {DM 2019-12-12}
+			$saved_tokens = \WC_Payment_Tokens::get_customer_tokens( $this->data['user_id'], $this->data['gateway_id'] );
+
+			foreach ( $saved_tokens as $saved_token ) {
+
+				if ( $saved_token->get_token() === $this->get_id() ) {
+					$token = $saved_token;
+					break;
+				}
+			}
+
+			if ( empty( $token ) ) {
+
+				// instantiate a new token
+				if ( $this->is_credit_card() ) {
+					$token = new \WC_Payment_Token_CC();
+				} elseif ( $this->is_echeck() ) {
+					$token = new \WC_Payment_Token_ECheck();
+				}
+
+				// mark the token as migrated
+				$this->data['migrated'] = true;
+
+				// update legacy token in user meta data
+				$gateways   = WC()->payment_gateways->payment_gateways();
+				// TODO: use `get_gateway_id`, once merged {DM 2019-12-12}
+				$gateway_id = $this->data['gateway_id'];
+
+				if ( ! empty( $gateway_id ) && ! empty( $gateway = $gateways[ $gateway_id ] ) ) {
```

Comment:
uhm maybe should we check if `get_payment_tokens_handler()` exists for that gateway for sanity?

Reply:
see another note below about the method being called here, maybe this logic can be kept inside the helper itself